src/tests/unittests/configuration_helper/adapters/test_m4i_instrument_adapter.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import logging
+from unittest import TestCase
+from unittest.mock import MagicMock, call, patch
+
+from qilib.configuration_helper import InstrumentAdapterFactory
+from qilib.configuration_helper.adapters import M4iInstrumentAdapter
+from tests.test_data.m4i_snapshot import snapshot
+
+
+class TestM4iInstrumentAdapter(TestCase):
+
+    def setUp(self):
+        sys.modules['pyspcm'] = MagicMock()
+        from qcodes.instrument_drivers.Spectrum.M4i import M4i
+
+        self.address = 'spcm1234'
+        self.adapter_name = 'M4iInstrumentAdapter'
+        self.instrument_name = f'{self.adapter_name}_{self.address}'
+        InstrumentAdapterFactory.instrument_adapters.clear()
+
+    def tearDown(self):
+        del sys.modules['pyspcm']
```

Comment:
If the `tearDown` is not called, the state of the system is not good. Perhaps use something like https://github.com/QCoDeS/Qcodes_contrib_drivers/pull/26/files#diff-c4454fdfa7ac2cf29ce0a0116d3885d2R13 in the test methods themself?

Reply:
you could also do this ones, at the top of the module:
```
sys.modules['pyspcm'] = MagicMock()
from qilib.configuration_helper import InstrumentAdapterFactory
del sys.modules['pyspcm']
```
I think

Reply:
Update the import to the top of the module.